test-xml-parsing/src/test/java/org/sonarsource/analyzer/commons/xml/checks/SonarXmlCheckVerifierTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * SonarSource Analyzers XML Parsing Test Commons
+ * Copyright (C) 2009-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.analyzer.commons.xml.checks;
+
+import org.assertj.core.util.Lists;
+import org.junit.Assert;
+import org.junit.Test;
+import org.sonarsource.analyzer.commons.xml.XmlFile;
+import org.w3c.dom.Element;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SonarXmlCheckVerifierTest {
+
+  @Test
+  public void test() {
+    SonarXmlCheckVerifier.verifyNoIssue("file.xml", new SilentTestCheck());
+    SonarXmlCheckVerifier.verifyIssueOnFile("file.xml", new FileTestCheck(), "Test file level message", 1, 2);
+    SonarXmlCheckVerifier.verifyIssues("checkTestFile.xml", new TestCheck());
+  }
+
+  @Test
+  public void missing_file() {
+    SilentTestCheck check = new SilentTestCheck();
+
+    IllegalStateException expected = Assert.assertThrows(IllegalStateException.class, () -> SonarXmlCheckVerifier.verifyIssueOnFile("missingFile.xml", check, "expected"));
+    assertThat(expected)
+      .hasMessageStartingWith("Unable to load content of file ")
+      .hasMessageEndingWith("missingFile.xml");
```

Comment:
Minor. I'm not sure why we mix assertj and junit here. Why not the following?
```suggestion
    Assertions.assertThatThrownBy(() -> SonarXmlCheckVerifier.verifyIssueOnFile("missingFile.xml", check, "expected"))
      .isInstanceOf(IllegalStateException.class)
      .hasMessageStartingWith("Unable to load content of file ")
      .hasMessageEndingWith("missingFile.xml");

```
Same thing for other test methods.

Reply:
good point, I somehow lost myself in the middle of the rewritting.